### PR TITLE
gce: Add support for bastions

### PR DESCRIPTION
**Pull request by hakman**
Closes https://github.com/kubernetes/kops/issues/15590
More fixes coming as part of https://github.com/kubernetes/kops/pull/15332

/cc @justinsb 
/assign @justinsb

### CODE REVIEW

File: pkg/model/gcemodel/firewall.go [hunk 1]
```diff
@@ -180,23 +180,27 @@ func (b *GCEModelContext) AddFirewallRulesTasks(c *fi.CloudupModelBuilderContext
...
 
 	ipv6 := *rule
 	ipv6.Name = s(b.NameForFirewallRule(name + "-ipv6"))
 	ipv6.Family = gcetasks.AddressFamilyIPv6
-	ipv6.SourceRanges = ipv6SourceRanges
-	if len(ipv6.SourceRanges) == 0 {
-		// We specify explicitly so the rule is in IPv6 mode
-		ipv6.Disabled = true
-		ipv6.SourceRanges = []string{"::/0"}
+	if len(ipv4.SourceTags) == 0 {
```
Comment (justinsb) [inline pkg/model/gcemodel/firewall.go:197]:
```suggestion
	if len(ipv6.SourceTags) == 0 {
```

**Comment by justinsb** [comment]
One nit with ipv6, but lgtm otherwise

/approve
/lgtm
/hold for nit

**Comment by hakman** [comment]
Thanks @justinsb. Nit fixed.

**Pull request merged**